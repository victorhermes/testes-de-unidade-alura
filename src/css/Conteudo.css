.container {
    display: flex;
    flex-direction: row;
    margin: 50px 20px 0 20px;
    border-bottom: 1px solid #a1a1a1;
    padding-bottom: 50px;
    width: 500px;
}

.imagem {
    width: 150px;
    height: 150px;
    border-radius: 5px;
    object-fit: cover;
}

.conteudo {
    margin-left: 40px;
}

.titulo-conteudo {
    font-size: 26px;
    margin-bottom: 0;
}

.botao-conferir {
    color: #01639c;
    font-size: 17px;
    font-weight: 600;
    border: 1px solid;
    padding: 4px 10px 4px 10px;
    border-radius: 5px;
}

.valor {
    font-size: 20px;
    color: #02980e;
    margin-bottom: 25px;
}
