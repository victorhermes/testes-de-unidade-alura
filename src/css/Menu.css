.menu {
    background-color: #2a5eec;
    height: 70px;
    top: 0;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
}

.menu-logo {
    width: 75px;
}
