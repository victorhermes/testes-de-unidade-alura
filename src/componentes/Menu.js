import React, { Component } from "react";
import { Link } from "react-router-dom";
import "../css/Menu.css";
import Logo from "../imgs/alura.svg";

export default class Menu extends Component {
    render() {
        return (
            <header className="menu">
                <Link to="/">
                    <img src={Logo} alt="Logo" className="menu-logo" />
                </Link>
            </header>
        );
    }
}
