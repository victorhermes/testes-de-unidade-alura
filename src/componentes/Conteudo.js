import React, { Component, Fragment } from "react";
import { URLBase } from "../services/api";

import Menu from "./Menu";
import ListarProdutos from "./ListaProdutos";

import "../css/Conteudo.css";

export default class Conteudo extends Component {
    state = {
        moeda: "R$",
        dados: []
    };

    componentDidMount() {
        fetch(URLBase)
            .then(response => response.json())
            .then(data => {
                this.setState({ dados: data });
            });
    }

    render() {
        return (
            <Fragment>
                <Menu />
                <ListarProdutos dados={this.state.dados} />
            </Fragment>
        );
    }
}
