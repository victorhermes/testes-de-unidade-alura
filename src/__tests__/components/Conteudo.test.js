import { shallow, mount } from "enzyme";
import React from "react";

import Conteudo from "../../componentes/Conteudo";
import Menu from "../../componentes/Menu";
import ListarProdutos from "../../componentes/ListarProdutos";

describe("Teste da renderização do componente Conteudo", () => {
    it("Deve haver o componente Conteudo", () => {
        const wrapper = shallow(<Conteudo />);

        expect(wrapper.exists()).toBe(true);
    });

    it("Deve renderizar o componente Menu", () => {
        const wrapper = shallow(<Conteudo />);

        expect(wrapper.find(Menu)).toHaveLength(1);
    });

    it("Deve renderizar o componente ListarProdutos", () => {
        const wrapper = shallow(<Conteudo />);

        expect(wrapper.find(ListarProdutos)).toHaveLength(1);
    });
});
