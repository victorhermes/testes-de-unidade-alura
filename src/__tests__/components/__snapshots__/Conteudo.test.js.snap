// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste da renderização do componente Conteudo Deve renderizar o componente Conteudo 1`] = `ShallowWrapper {}`;
